Client/Frontend/Browser/TabManager.swift

```diff
@@ -1082,6 +1084,9 @@ extension TabManager: PreferencesObserver {
             }
             // The default tab configurations also need to change.
             configuration.preferences.javaScriptCanOpenWindowsAutomatically = allowPopups
+        case Preferences.General.mediaAutoPlays.key:
+            let allowsMediaAutoPlay = Preferences.General.mediaAutoPlays.value
+            configuration.mediaTypesRequiringUserActionForPlayback = allowsMediaAutoPlay ? [] : .all
```

Comment:
Do we need to update all the current tabs? (see blockPopups update above)

Reply:
@kylehickinson  We can't change WKWebViewConfiguration after the web-view has been created :(

Reply:
Ah, so we'd have to destroy all web views on change if we wanted that to fix current tabs? Same as cookie control I guess, probably not worth it? @iccub What do you think?

Reply:
I think we should recreate all web views, we do that already for another preference `blockAllCookies`

See here https://github.com/brave/brave-ios/blob/development/Client/Frontend/Browser/BrowserViewController.swift#L3026

Reply:
Hmm but why re-create all of them for `auto-play`? I mean, I have no problem doing it, I just don't understand why we want to do that.

If the user wants to play the video, they'd tap it no?
I ask because in the future this will affect background play. If I reload all tabs in the background and each tab starts playing it will be really weird..

Reply:
It's a minor thing so not recreating it is fine too :) especially if you predict unexpected behavior in the future

Reply:
FYI we don't need to reload all the tabs, just destroy their web views so that the next time they are opened they will reload automatically (IIRC)

Reply:
@kylehickinson I tried that. They don't reload automatically unless you call `loadPrivileged request`

Reply:
@Brandon-T Doing this here is not required since you are doing `reset()` below

Reply:
@kylehickinson I fixed it. It will lazy reload now. Only the selected tab reloads and other tabs will reload only when you tap on them.